phpBB/posting.php

```diff
@@ -1012,6 +1012,28 @@
 	// Also check if subject got updated...
 	$update_subject = $mode != 'edit' || ($post_data['post_subject_md5'] && $post_data['post_subject_md5'] != md5($post_data['post_subject']));
 
+	/* Send the posted text to parser to decode the @mentions.
+	   Convert each mention into links to corresponding user profiles
+	   Push notification to the mentioned users.*/
+	$helper_container = $phpbb_container->get('phpbb_mention_helper');
+	$post_parsing_data = $helper_container->get_mentioned_users($message_parser->message);
+
+	if (is_array($post_parsing_data) &&
```

Comment:
I think it would make more sense to have the get_mentioned_users return always the same datatype, that way you never need to check like you do now.